### PR TITLE
Release v0.1.5

### PR DESCRIPTION
**Pull request by detiber**
**What this PR does / why we need it**:

Updates the release-0.1 branch for the v0.1.5 release

**Release note**:
```release-note
NONE
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -22,7 +22,7 @@ export KUBEBUILDER_CONTROLPLANE_START_TIMEOUT ?=60s
 export KUBEBUILDER_CONTROLPLANE_STOP_TIMEOUT ?=60s
 
 # Image URL to use all building/pushing image targets
-export CONTROLLER_IMG ?= gcr.io/k8s-cluster-api/cluster-api-controller:v0.1.4
+export CONTROLLER_IMG ?= gcr.io/k8s-staging-cluster-api/cluster-api-controller:v0.1.5
```
Comment (detiber) [inline Makefile]: This updates the Makefile to push to the staging repo by default.

File: config/default/manager_image_patch.yaml [hunk 2]
```diff
@@ -7,5 +7,5 @@ spec:
   template:
     spec:
       containers:
-      - image: gcr.io/k8s-cluster-api/cluster-api-controller:v0.1.4
+      - image: k8s.gcr.io/cluster-api/cluster-api-controller:v0.1.5
```
Comment (detiber) [inline config/default/manager_image_patch.yaml]: This *should* be the home of the promoted image after running through the Image Promotion process.

File: docs/developer/releasing.md [hunk 3]
```diff
@@ -40,8 +45,10 @@ For version v0.x.y:
    images and fat manifest with `REGISTRY="gcr.io/k8s-cluster-api" make all-push` (on the 0.1 release branch, we
```
Comment (ncdc) [inline docs/developer/releasing.md]: I think this needs adjusting?

File: docs/developer/releasing.md [hunk 4]
```diff
@@ -40,8 +45,10 @@ For version v0.x.y:
...
    do `make docker-push`)
 7. Create an annotated tag from this same commit `git tag -a v0.x.y -m v0.x.y` and push the tag to the github repository `git push origin v0.x.y`
```
Comment (ncdc) [inline docs/developer/releasing.md]: I know this isn't new in this PR, but shouldn't we create the tag **before** building the image? I'm guessing if there's any code that sets ldflags for version based on `git describe` output, we'd need the tag there first.

Comment (detiber) [reply docs/developer/releasing.md]: done

File: docs/developer/releasing.md (hunk 3 above)
Comment (detiber) [reply docs/developer/releasing.md]: It applies to master rather than the `release-0.1` branch, but I went ahead and updated here as well.

**Comment by detiber** [comment]
I'll need to sync the releasing doc changes back to `master`, but I'll do that once I've finished confirming all the steps.

**Comment by ncdc** [comment]
/lgtm
/hold
/assign @chuckha 

Chuck - please 👀 and unhold if you're happy w/it. Thanks!

**Comment by chuckha** [comment]
this looks great

/hold cancel

**Pull request merged**